subprojects/launcher/src/test/groovy/org/gradle/launcher/daemon/configuration/BuildProcessTest.groovy

```diff
@@ -237,12 +239,42 @@ class BuildProcessTest extends Specification {
         def parametersWithDefaults = buildParameters(DaemonParameters.DEFAULT_JVM_ARGS)
 
         when:
-        def buildProcess = new BuildProcess(currentJvm, new JvmOptions(fileCollectionFactory))
+        def buildProcess = createBuildProcess()
 
         then:
         !buildProcess.configureForBuild(parametersWithDefaults)
     }
 
+    def "instrumentation agent status is immutable"() {
+        given:
+        def desiredParameters = buildParameters([])
+        desiredParameters.setApplyInstrumentationAgent(desiredStatus)
+
+        when:
+        def currentAgentStatus = Stub(AgentStatus) {
+            isAgentInstrumentationEnabled() >> currentStatus
+        }
+        BuildProcess buildProcess = createBuildProcess(currentAgentStatus)
+
+        then:
+        buildProcess.configureForBuild(desiredParameters) == expectedResult
+
+        where:
+        desiredStatus | currentStatus || expectedResult
+        false         | false         || true
```

Comment:
❓ This truth table should include comments describing _why_ each result is expected. I couldn't make sense of it.